components/service/experiments/src/test/java/mozilla/components/service/experiments/ExperimentsTest.kt

```diff
@@ -77,14 +81,21 @@ class ExperimentsTest {
         `when`(experimentsUpdater.getExperimentSource(configuration)).thenReturn(experimentSource)
     }
 
+    private fun mockUpdaterWorkManagerFunctionality() {
+        // Since WorkManager doesn't support working with robolectric, need to mock running the task
+        // but doing it synchronously to avoid a lot of headaches, to guarantee we don't need to
+        // await anything.
+        `when`(experimentsUpdater.scheduleUpdates()).then {
```

Comment:
I believe it could, I will refactor and see.  There were several tests that didn't require it (i.e. they pass without it) which is why I split it out like this.